Libraries/Network/XMLHttpRequest.js

```diff
@@ -423,12 +423,46 @@ class XMLHttpRequest extends (EventTarget(...XHR_EVENTS): any) {
       // according to the spec, return null if no response has been received
       return null;
     }
-    const headers = this.responseHeaders || {};
-    return Object.keys(headers)
-      .map(headerName => {
-        return headerName + ': ' + headers[headerName];
-      })
-      .join('\r\n');
+
+    // Combine headers by lower case into lists first.
+    const lowerCaseHeaders = {};
+    Object.keys(this.responseHeaders).forEach(headerName => {
+      const value = this.responseHeaders[headerName];
+      const lowerCaseHeaderName = headerName.toLowerCase();
+      const values = lowerCaseHeaders[lowerCaseHeaderName];
+      if (values) {
+        values.push(value);
+        return;
+      }
+      lowerCaseHeaders[lowerCaseHeaderName] = [value];
+    });
+
+    // Combine lists of headers into comma-and-space-separated values.
+    const combinedHeaders = Object.keys(lowerCaseHeaders).map(headerName => {
+      return [headerName, lowerCaseHeaders[headerName].join(', ')];
+    });
```

Comment:
Instead of allocating arrays for the edge case where there are multiple, differently-cased values for the same header, what do you think about keeping it simple and appending strings as needed?

```suggestion
// Assign to non-nullable local variable.
const responseHeaders = this.responseHeaders;

const unsortedHeaders = {};
for (const rawHeaderName of Object.keys(responseHeaders)) {
  const headerValue = responseHeaders[rawHeaderName];
  const lowerHeaderName = rawHeaderName.toLowerCase();
  if (unsortedHeaders.hasOwnProperty(lowerHeaderName)) {
    unsortedHeaders[lowerHeaderName] += ', ' + headerValue;
  } else {
    unsortedHeaders[lowerHeaderName] = headerValue;
  }
}
```